docs/concepts/dependencies.md

```diff
@@ -1,13 +1,29 @@
 # Specifying dependencies
 
-In uv, project dependency specification is divided between two `pyproject.toml` tables:
-`project.dependencies` and `tool.uv.sources`.
+In uv, project dependencies are declared across two `pyproject.toml` tables: `project.dependencies`
+and `tool.uv.sources`.
 
-`project.dependencies` is used to define the standards-compliant dependency metadata, propagated
-when uploading to PyPI or building a wheel. `tool.uv.sources` is used to specify the _sources_
-required to install the dependencies, which can come from a Git repository, a URL, a local path, a
-different index, etc. This metadata must be expressed separately because the `project.dependencies`
-standard does not allow these common patterns.
+`project.dependencies` defines the standards-compliant dependency metadata, propagated when
+uploading to PyPI or building a wheel.
+
+`tool.uv.sources` enriches the dependency metadata with additional sources, incorporated during
+development. A dependency source can be a Git repository, a URL, a local path, or an alternative
+registry.
+
+`tool.uv.sources` enables uv to support common patterns like editable installations and relative
+paths that are not supported by the `project.dependencies` standard. For example:
+
+```toml title="pyproject.toml"
+[project]
+name = "albatross"
+version = "0.1.0"
+dependencies = [
+  "bird-feeder",
+]
+
+[tool.uv.sources]
+bird-feeder = { path = "/path/to/bird-feeder" }
+```
```

Comment:
This might be a bit much... But I wanted to convey the idea succinctly even for those that won't read the full page.